File: Data Structures & Algorithms/is-anagram/submission-3.js
class Solution {
    /**
     * @param {string} s
     * @param {string} t
     * @return {boolean}
     */
    isAnagram(s, t) {

        if (s.length != t.length)
            return false;

        const Count = {};

        for(let char of s){
            Count[char] = (Count[char] || 0) + 1;
        }  

        for(let char of t){
            if(!Count[char]){
                return false;
            }
            Count[char]--;
        }
        return true;
    }
}

File: Data Structures & Algorithms/duplicate-integer/submission-3.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        
        std::unordered_set<int> numbers;

        for( int num : nums){
            if(numbers.count(num))
                return true;
            else 
                numbers.insert(num);        
        }
        return false;
    }
};
